### PR TITLE
should you get it here bug fix

### PR DESCRIPTION
**Pull request by JPrevost**
## Status
**READY**

#### What does this PR do?

Decouples ILL and recall/hold logic and hopefully doesn't create more bugs than it fixes.

#### Helpful background context (if appropriate)

It probably creates more bugs than it fixes. (Melissa checked the PR build and signed off that this _ seems_ to not introduce bugs)

#### How can a reviewer manually see the effects of these changes?

See ticket for test records.

#### What are the relevant tickets?
- https://mitlibraries.atlassian.net/browse/DI-660

#### Todo:
- [ ] Tests
- [ ] Documentation
- [x] Stakeholder approval

#### Requires Database Migrations?
NO

#### Includes new or updated dependencies?
NO

### CODE REVIEW

File: app/models/button_maker.rb [hunk 1]
```diff
@@ -15,6 +15,8 @@
 # Refactor, but it just got too janky to deal with all the edge cases in our
 # circ logic - too hard to tell whether they'd all been dealt with, except by
 # pulling it all into a special-purpose, encapsulated place.
+# You probably don't want to, but when all else fails familiarize yourself with:
```
Comment (thatandromeda) [inline app/models/button_maker.rb]: ++ for adding this link

**Comment by JPrevost** [comment]
tldr; I'll take care of the better name very soon in a separate PR :)

I've got `should_you_get_it_here?` renamed `available_locally?` right now in my bigger refactor that will follow this.  `currently_on_the_shelves?` seems good too. Let's discuss that in the follow-on PR. It's coming along nicely and I'm just (carefully) moving tests around to match the new models before opening that one up for review.

**Comment by thatandromeda** [comment]
Sounds good. :shipit:
